### PR TITLE
feat(VBC-123): load guest wishlist

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
Load action for the guest wishlist

## Description
Implementation of load action for the guest wishlist 

## Related Issue
VBC-123

## How Has This Been Tested?

Unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by darocha — APPROVED** [review]
No change requests from me. It's ok to merge.

File: packages/composables/__mocks__/localStorage.mock.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      store[key] = value.toString();
+    },
+    clear: () => {
+      store = {};
+    },
+    removeItem: (key: string) => {
+      delete store[key];
+    }
+  };
+})();
```
Comment (zoltantakacs-born) [inline packages/composables/__mocks__/localStorage.mock.ts:17]: This is not used.

File: packages/composables/src/useGuestWishlist/actions/load.ts [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
 
+export const load = async (context: Context): Promise<GuestWishlist> => {
+  let guestWishlist: GuestWishlist = {
+    name: 'Guest Wishlist',
+    items: [],
+    wishlist_product_data: { data: [], meta: {} },
+    is_public: true
+  };
+
+  const localStorageItem = localStorage.getItem('bigcommerce-wishlist');
```
Comment (zoltantakacs-born) [inline packages/composables/src/useGuestWishlist/actions/load.ts:11]: Could you please store the name of the localstorage key in a contant and use that in here and unit tests as well?

File: packages/composables/src/useGuestWishlist/actions/load.ts [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+
+  if (!localStorageItem) {
+    localStorage.setItem('bigcommerce-wishlist', JSON.stringify(guestWishlist));
+    return guestWishlist;
+  }
+
+  guestWishlist = JSON.parse(localStorageItem);
```
Comment (zoltantakacs-born) [inline packages/composables/src/useGuestWishlist/actions/load.ts:18]: some error handling with try catch would be nice

**Review by zoltantakacs-born — CHANGES_REQUESTED** [review]

File: packages/composables/src/useGuestWishlist/actions/load.ts (hunk 3 above)
Comment (WojtekTheWebDev) [reply packages/composables/src/useGuestWishlist/actions/load.ts:19]: Error handling is in the `useGuestWishlist`'s `load` method:
Screenshot 2021-11-30 at 16 49 31

**Review by zoltantakacs-born — APPROVED** [review]

**Pull request merged**